socket.go

```diff
@@ -286,6 +287,12 @@ func (sck *socket) addConn(c *Conn) {
 	if sck.r != nil {
 		sck.r.addConn(c)
 	}
+	// resend subscriptions for topics if there are any
+	if sck.subTopics != nil {
+		for _, topic := range sck.subTopics() {
+			_ = sck.Send(NewMsg(append([]byte{1}, topic...)))
+		}
+	}
 	sck.mu.Unlock()
```

Comment:
perhaps move this up as a `defer sck.mu.Unlock()`

Reply:
done